base_layer/core/src/transactions/weight.rs

```diff
@@ -33,7 +33,7 @@ pub struct WeightParams {
     /// Weight in grams per output, excl. TariScript and OutputFeatures
     pub output_weight: u64,
     /// Metadata per byte weight
-    pub metadata_bytes_per_gram: Option<NonZeroU64>,
+    pub metadata_bytes_per_gram: Option<NonZeroU64>, // Why is this defined like this? Seems very hacky
```

Comment:
Why?

Reply:
I believe the option was necessary in the past because dibbler did not have this metadata weighting. It can be simplified now.

Reply:
This is moot anyway, since the coinbase does not charge fees and is effectively free. Since it has a max, there is no need weight it specially, and in fact we want to encourage miners to use this field

Reply:
Added #4997

Reply:
Yup agree, no need to use up the total block weight for coinbase extra field. metadata weighting refers to all UTXO metadata (e.g. sidechain features etc) not just that single field.